packages/grafana-ui/src/components/uPlot/Plot.tsx

```diff
@@ -36,7 +39,19 @@ export class UPlotChart extends React.Component<PlotProps, UPlotChartState> {
     this.state = {
       ctx: {
         plot: null,
-        getCanvasBoundingBox: () => this.plotCanvasBBox.current,
+        getCanvasBoundingBox: () => {
+          // If cursor sync enabled poll canvas bbox once per CANVAS_POLL_DELAY
+          if (this.props.config.hasSync() && performance.now() - this.canvasPollTs > CANVAS_POLL_DELAY) {
+            let canvas = this.plotContainer.current?.querySelector<HTMLDivElement>('.u-over');
+            if (!canvas) {
+              return null;
+            }
+
+            (this.plotCanvasBBox as MutableRefObject<any>).current = canvas.getBoundingClientRect();
+            this.canvasPollTs = performance.now();
+          }
+          return this.plotCanvasBBox.current;
```

Comment:
the ideal solution would be if we could invalidate the bounding box on dashboard scroll but how to know if the dashboard has scrolled or not here is the challange, can't think of an easy way to do that :(

Reply:
uPlot internally uses two, global, capturing-phase `window.scroll` and `window.resize` listeners to set invalidation flags on its internal registry of all created uPlot instances. then on mousemove events, each chart re-syncs its rect prior to doing any additional math, if it has been invalidated. perhaps a similar strategy can be used here.

Reply:
@dprokop i added a `syncRect` hook to uPlot in https://github.com/leeoniya/uPlot/commit/d9fcf5d3300670340a1d6576de48158d73291bbc which will fire when it's necessary to update the coords. i tested it with uPlot's native sync and it fires correctly on all plots of first mousemove/mousenter after resize or scroll.

this may simplify this PR a lot, or not at all.

Reply:
@leeoniya thanks, let me check this out!

Reply:
@leeoniya the synRect hooks works like a charm, awesome! When do you plan to release ne uPlot version? I see this is not available in any relase yet, but would be awesome to get that in beta3